src/lib/normalise-dag-node.js

```diff
@@ -33,65 +121,96 @@ export function normaliseDagPb (node, cid, type) {
   // if (toCidStrOrNull(node.multihash) !== cid) {
   //   throw new Error('dag-pb multihash should match provided cid')
   // }
-  node = node.toJSON()
-  let format
+
+  const nodeAsJson = convertDagPbNodeToJson(node)
+  /**
+   * @type {NormalizedDagPbNodeFormat}
+   */
+  let format = 'unknown'
   try {
     // it's a unix system?
-    const { type, data, blockSizes } = UnixFS.unmarshal(node.data)
-    node.data = { type, data, blockSizes }
+    const { type, data, blockSizes } = UnixFS.unmarshal(nodeAsJson.data)
+    console.log('type, data, blockSizes: ', type, data, blockSizes)
+    nodeAsJson.data = { type, data, blockSizes }
     format = 'unixfs'
   } catch (err) {
     // dag-pb but not a unixfs.
     // console.log(err)
   }
+  const cidStr = toCidStrOrNull(cid)
+  if (cidStr == null) {
+    throw new Error(`cidStr is null for cid: ${cid}`)
+  }
 
   return {
-    cid,
+    cid: cidStr,
     type,
-    data: node.data,
-    links: normaliseDagPbLinks(node, cid),
-    size: node.size,
+    data: nodeAsJson.data,
+    links: normaliseDagPbLinks(nodeAsJson, cid),
+    size: nodeAsJson.size,
     format
   }
 }
 
-/*
+/**
  * Convert DagLink shape into normalized form that can be used interchangeably
  * with links found in dag-cbor
+ *
+ * @param {DagPbNodeAsJson} node
+ * @param {string} sourceCid
+ * @returns {NormalizedDagLink[]}
  */
 export function normaliseDagPbLinks (node, sourceCid) {
-  return node.links.map(({ name, size, cid }, index) => ({
-    path: name || `Links/${index}`,
-    source: sourceCid,
-    target: cid,
-    size,
-    index
-  }))
+  return node.links.map((link, index) => {
+    const { Name: name, Tsize: size, Hash: cid } = link
+
+    return ({
+      path: name || `Links/${index}`,
+      source: sourceCid,
+      target: cid.toString(),
+      size: size ?? 0,
+      index
+    })
+  })
 }
 
-/*
+/**
  * Find links and add type and cid info
+ *
+ * @function normaliseDagCbor
+ * @param {unknown} obj - The data object
+ * @param {string} cid - The string representation of the CID
+ * @param {number} code - multicodec code, see https://github.com/multiformats/multicodec/blob/master/table.csv
+ * @returns {NormalizedDagNode}
  */
-export function normaliseDagCbor (obj, cid, type) {
+export function normaliseDagCbor (obj, cid, code) {
   const links = findAndReplaceDagCborLinks(obj, cid)
   return {
     cid,
-    type,
+    type: code,
     data: obj,
-    links: links
+    links,
+    size: links.reduce((acc, { size }) => acc + size, 0),
+    format: 'unknown'
   }
 }
 
 /**
  * A valid IPLD link in a dag-cbor node is an object with single "/" property.
+ *
+ * @param {unknown} obj
+ * @param {string} sourceCid
+ * @param {string} path
+ * @returns {NormalizedDagLink[]}
  */
 export function findAndReplaceDagCborLinks (obj, sourceCid, path = '') {
```

Comment:
This function is a bit of a mess (not your mess, this is ancient stuff), there's also the fact that your fix for the `'object'` check is going to avoid arrays now, so the bug was hiding issues that already exists.

We have this logic tested and used in js-multiformats, so maybe you could use that instead? if you import `multiformats/block' then you could use `links()` on a block form of a decoded object, and that would cover any object type regardless of codec (the fact that this is dedicated to dag-cbor is a bit weird since `obj` could be anything—probably a historical thing but the logic in here seems to differentiate dag-pb from everything else, which is the appropriate way to deal with dag-pb since it's legacy and everything else is newer).

See https://github.com/multiformats/js-multiformats/blob/master/test/test-block.js#L45-L51 for this in action, it'll properly walk through the object and yield the links in the object along with the paths. You could wrap that in the additional stuff you need to get your `size` etc.

Alternatively, maybe steal some of the logic in there because we've worked on that one (and `tree()` to better handle all possible node types and iterate through them - bytes, null, arrays, CIDs, they all come with special cases which is annoying to have to reimplement and test for yourself. https://github.com/multiformats/js-multiformats/blob/f2664fbd4dd67b9b3f329bf827aa2439ed1a4a81/src/block.js#L49-L68

Reply:
I'm going to defer that to issue https://github.com/ipfs/ipld-explorer-components/issues/338